### PR TITLE
X697

### PR DESCRIPTION
**Pull request by jbeast**
New stain types.

### CODE REVIEW

File: src/pages/staining/ComplexStainForm.tsx [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    panel: Yup.string()
+      .oneOf(Object.values(StainPanel))
+      .required()
+      .label("Experimental Panel"),
+    labware: Yup.array()
+      .of(
+        Yup.object().shape({
+          barcode: Yup.string().required().label("Barcode"),
+          bondBarcode: Yup.string().required().label("Bond Barcode"),
+          bondRun: Yup.number().integer().positive().label("Bond Run"),
```
Comment (seenanair) [inline src/pages/staining/ComplexStainForm.tsx:85]: In acceptance test it says -bond run number - 4 digit integer (+ve) . Is it required  to put 4 digit number constraint here?

**Review by seenanair — COMMENTED** [review]
Also, in Acceptance test it says- Page will allow you to submit a request, then change the stain type (without losing the other information entered)

File: src/pages/staining/ComplexStainForm.tsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+  const validationSchema = Yup.object().shape({
+    stainType: Yup.string().required().label("Stain Type"),
+    plex: Yup.number()
+      .integer()
+      .min(plexMin)
+      .max(plexMax)
+      .required()
+      .label("Plex Number"),
+    panel: Yup.string()
+      .oneOf(Object.values(StainPanel))
```
[the remark this answers is not in the document]
Comment (jbeast) [reply src/pages/staining/ComplexStainForm.tsx:77]: Good point, well spotted. I'll change.

File: src/pages/staining/ComplexStainForm.tsx (hunk 1 above)
Comment (jbeast) [reply src/pages/staining/ComplexStainForm.tsx:85]: I thought that might be a bit too restrictive for the client. Seemed like the sort of thing that may need to change in the relatively near future, so thought I'd leave that as core side validation only.

**Pull request merged**